### PR TITLE
[google_maps_flutter_platform_interface] Platform interface changes for #3258

### PR DESCRIPTION
**Pull request by Rexios80**
See #3258

### CODE REVIEW

File: packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/map_configuration.dart [hunk 1]
```diff
@@ -3,8 +3,7 @@
 // found in the LICENSE file.
 
 import 'package:flutter/widgets.dart';
-
-import 'ui.dart';
+import '../../google_maps_flutter_platform_interface.dart';
```
Comment (stuartmorgan) [inline packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/map_configuration.dart:6]: Nit: Our style is to leave a blank line between package and non-package dependencies.

File: packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/map_configuration.dart [hunk 2]
```diff
@@ -48,25 +54,32 @@ class MapConfiguration {
...
 
   /// True if scroll gestures should be enabled.
+  ///
+  /// Android/iOS only. For web, see [webGestureHandling].
   final bool? scrollGesturesEnabled;
 
   /// True if tilt gestures should be enabled.
   final bool? tiltGesturesEnabled;
 
+  /// True if 45 degree imagery should be enabled. Web only.
```
Comment (stuartmorgan) [inline packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/map_configuration.dart:71]: Nit: "Web only." should be its own paragraph (like the comparable comments you're adding above), per Effective Dart style.

**Review by stuartmorgan — CHANGES_REQUESTED** [review]
The `applyDiff` tests look good, thanks! There should be tests for equality (that changing just each of the new properties causes objects not to be equal), hash (that changing just each of the new properties changes the hash) and `diffFrom` as well though.

File: packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/map_configuration.dart (hunk 2 above)
Comment (Rexios80) [reply packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/map_configuration.dart:71]: Done

File: packages/google_maps_flutter/google_maps_flutter_platform_interface/CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,6 @@
 ## 2.3.0
 
+* Adds options for gesture handling and tilt controls on web.
```
Comment (stuartmorgan) [inline packages/google_maps_flutter/google_maps_flutter_platform_interface/CHANGELOG.md:3]: Looks like you accidentally merged into another version update on the last update.

**Review by stuartmorgan — APPROVED** [review]
LGTM once the version merge is fixed, thanks!

File: packages/google_maps_flutter/google_maps_flutter_platform_interface/CHANGELOG.md (hunk 3 above)
Comment (Rexios80) [reply packages/google_maps_flutter/google_maps_flutter_platform_interface/CHANGELOG.md:3]: Although it looks like the pubspec version change is missing?

Comment (Rexios80) [reply packages/google_maps_flutter/google_maps_flutter_platform_interface/CHANGELOG.md:3]: Okay I think I fixed it

File: packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/map_configuration.dart [hunk 4]
```diff
@@ -4,7 +4,7 @@
 
 import 'package:flutter/widgets.dart';
 
-import 'ui.dart';
```
Comment (ditman) [inline packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/map_configuration.dart:7]: Removing `import 'ui.dart';` is great, thanks!

**Review by ditman — APPROVED** [review]
LGTM!

**Pull request merged**